File: web_app/types/comment.ts
import { User } from './user';

export interface Comment {
    id: string;
    taskId: string;
    userId: string;
    content: string;
    parentId?: string;
    images?: string[];
    createdAt: string;
    updatedAt: string;
    user?: User;
    children?: Comment[];
}

File: auction_system/backend/internal/models/comment.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Comment struct {
	ID        uint           `gorm:"primaryKey" json:"id"`
	AuctionID uint           `gorm:"not null;index" json:"auction_id"`
	Auction   Auction        `json:"-"` // Avoid circular dependency in JSON or just omit
	UserID    uint           `gorm:"not null" json:"user_id"`
	User      User           `json:"user"`
	Content   string         `gorm:"type:text;not null" json:"content"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}
